src/components/ProductCard.tsx

```diff
@@ -1,7 +1,26 @@
+import { useFavorites } from "../hooks/useFavorites";
 import { icons } from "./common";
-import { Product } from "~/api/products";
+import { Product } from "../api/products";
+import { useFavoritesStore } from "../stores";
 
 export const ProductCard = ({ product }: { product: Product }) => {
+  const favoritesStore = useFavoritesStore();
+  const isProductInFavorites = useFavoritesStore().favorites.some(
+    (favProduct) => favProduct.id === product.id
+  );
+
+  const handleHeartClick = () => {
+    if (isProductInFavorites) {
+      const updatedFavorites = favoritesStore.favorites.filter(
+        (favProduct) => favProduct.id !== product.id
+      );
+      favoritesStore.setFavorites(updatedFavorites);
+    } else {
+      const updatedFavorites = [...favoritesStore.favorites, product];
+      favoritesStore.setFavorites(updatedFavorites);
+    }
```

Comment:
When updating a state based on it's previous value, it's a good practice to do it using the parameter that comes with the setter (in this case, `setFavorites`).

This is because updates are asynchronous and the value you get from `favoritesStore.favorites` might not be what you expect.

Here is an example of the better way of doing it, for the `else`:

`setFavorites((prev) => [...prev, product])`

All this logic can actually be reduced to this:

```
setFavorites((prev) => isProductInFavorites 
    ? prev.filter((p) => p.id !== product.id)
    : [...prev, product]
)
```